src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -354,7 +392,7 @@ protected override void Go()
                 switch (_operator)
                 {
                     case RegexOpcode.Stop:
-                        return;
+                        return runmatch!._matchcount[0] > 0;
```

Comment:
We do this check in a few places.  Consider adding an internal property for it on Match.